keps/sig-scheduling/3370-native-podgroup-api/README.md

```diff
@@ -0,0 +1,1111 @@
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+Propose a native PodGroup API in Kubernetes to represent an enforced constraint to 
+schedule a group of pods altogether.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Kubernetes provides a number of scheduling directives to abstract workloads’ 
+placement requirements. Till now, they’re mostly resource requests or topological
+inter-pod hard/soft restrictions, and executed on a single pod’s basis, in a
+stateless manner. This works great for most workloads. However, if we look at the
+workload level, it’s not good enough because sometimes the entire workload needs
+to be scheduled altogether or do nothing.
+
+Lacking this "all-or-nothing" semantics can cause a workload to be partially
+scheduled. This, from an application’s perspective, leads to an undetermined SLO
+since the workload cannot be guaranteed to be ready within an expected period.
+Sometimes it can be worse: workloads may be partially scheduled simultaneously to
+reserve their resources and then cause deadlocks.
+
+This "all-or-nothing" semantics, also known as coscheduling or gang-scheduling, 
+is actually a typical requirement for batch (run-to-completion) workloads. We’re
+proposing a fundamental Kubernetes PodGroup API to achieve this. This new API,
+which served as a scheduling primitive, can be used along with other scheduling 
+building blocks, to accommodate more workloads to run in Kubernetes natively and
+effortlessly. Moreover, this enforced API will help achieve efficient cluster
+utilization and ease users to define their application SLOs.
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Define PodGroup API in Kubernetes (k/k)
+- Enforce PodGroup as a scheduling primitive
+- Manage the lifecycle of PodGroup
+- Compatible with CA (Cluster Autoscaler)
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Restrict the PodGroup implementation to a specific scheduler
+
```

Comment:
Let's add a bullet stating that PodGroup based preemption would be a non-gola for alpha implementation.

Reply:
Ok. Added it.

Reply:
But maybe we could add this to risks if pods  with high priority in a PodGroup preempt successfully but finally released for some reasons.